huaweicloud/services/as/data_source_huaweicloud_as_lifecycle_hooks.go

```diff
@@ -0,0 +1,150 @@
+package as
+
+import (
+	"context"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/go-uuid"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/chnsz/golangsdk/openstack/autoscaling/v1/lifecyclehooks"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+)
+
+var convertHookTypeMap = map[string]string{
+	"INSTANCE_LAUNCHING":   "ADD",
+	"INSTANCE_TERMINATING": "REMOVE",
+}
+
+// API: AS GET /autoscaling-api/v1/{project_id}/scaling_lifecycle_hook/{scaling_group_id}/list
+func DataSourceLifeCycleHooks() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceLifeCycleHooksRead,
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
```

Comment:
```suggestion
```
The `Computed` behavior is useless in the data source.
each operations (terraform refresh, terraform apply, terraform plan), the data source will execute a query, they are not doing update, and not display the change.